website/nifti-viewer/index.js

```diff
@@ -0,0 +1,232 @@
+cornerstoneNIFTIImageLoader.external.cornerstone = cornerstone;
+const ImageId = cornerstoneNIFTIImageLoader.nifti.ImageId;
+cornerstoneNIFTIImageLoader.nifti.streamingMode = true;
+const niftiReader = cornerstoneNIFTIImageLoader.external.niftiReader;
+
+let loaded = false;
+const synchronizer = new cornerstoneTools.Synchronizer("cornerstonenewimage", cornerstoneTools.updateImageSynchronizer);
+
+let voxel3dUnits = [0, 0, 0];
+let dims = [0, 0, 0, 0];
+let niftiImageBuffer = [];
+
+// Helper to calculate voxel position based on view
+function updateVoxelCoordinates(view, voxelCoords) {
+    if (view === 'axial') {
+        voxel3dUnits[0] = voxelCoords.x;
+        voxel3dUnits[1] = voxelCoords.y;
+    } else if (view === 'sagittal') {
+        voxel3dUnits[1] = voxelCoords.x;
+        voxel3dUnits[2] = voxelCoords.y;
+    } else if (view === 'coronal') {
+        voxel3dUnits[0] = voxelCoords.x;
+        voxel3dUnits[2] = voxelCoords.y;
+    }
+}
+
+// Event listener for mouse drag to update voxel coordinates
+function addMouseDragListener(element, view) {
+    element.addEventListener('cornerstonetoolsmousedrag', (event) => {
+        const voxelCoords = event.detail?.currentPoints?.image;
+        if (voxelCoords) {
+            updateVoxelCoordinates(view, voxelCoords);
+            handleVoxelClick(voxel3dUnits);
+        }
+    });
+}
+
+// Load and display NIfTI image on a specific element
+function loadAndViewImage(element, imageId, view) {
+    const imageIdObject = ImageId.fromURL(imageId);
+    element.dataset.imageId = imageIdObject.url;
+
+    cornerstone.loadAndCacheImage(imageIdObject.url).then(image => {
+        setupImageViewport(element, image);
+        setupImageTools(element, imageIdObject);
+
+        synchronizer.add(element);
+        addMouseDragListener(element, view);
+
+        element.addEventListener('cornerstonestackscroll', (event) => updateSliceIndex(view, event.detail.newImageIdIndex));
+    }).catch(err => {
+        console.error(`Error loading image for ${view} view:`, err);
+    });
+    element.addEventListener('click', function (event) {
+        //TODO: Update the clicked voxel information.
+        console.log(event)
+    });
+}
+
+// Setup viewport and display the image
+function setupImageViewport(element, image) {
+    const viewport = cornerstone.getDefaultViewportForImage(element, image);
+    cornerstone.displayImage(element, image, viewport);
+    cornerstone.resize(element, true);
+}
+
+// Enable tools and interactions for the displayed image
+function setupImageTools(element, imageIdObject) {
+    const numberOfSlices = cornerstone.metaData.get('multiFrameModule', imageIdObject.url).numberOfFrames;
+    const stack = {
+        currentImageIdIndex: imageIdObject.slice.index,
+        imageIds: Array.from({ length: numberOfSlices }, (_, i) => `nifti:${imageIdObject.filePath}#${imageIdObject.slice.dimension}-${i},t-0`)
+    };
+
+    cornerstoneTools.addStackStateManager(element, ['stack']);
```

Comment:
I think we also want a rescale tool, sometimes I can barely see anything in those images.

Reply:
You mean a tool for zooming in/out, right?

Reply:
No, brightness/contrast is what they seem to call it here. https://www.cornerstonejs.org/live-examples/videocolor

Reply:
got it, will work on that.